lib/stm32/h7/pwr.c

```diff
@@ -50,16 +114,49 @@ void pwr_set_svos_scale(enum pwr_svos_scale scale)
 }
 
 void pwr_set_vos_scale(enum pwr_vos_scale scale) {
-	rcc_periph_clock_enable(RCC_SYSCFG);  /* Ensure we can access ODEN. */
-	uint32_t d3cr_masked = PWR_D3CR & ~(PWR_D3CR_VOS_MASK << PWR_D3CR_VOS_SHIFT);
+	static const uint8_t srdcr_vos_values[] = {
+		PWR_SRDCR_VOS_SCALE_0,
+		PWR_SRDCR_VOS_SCALE_1,
+		PWR_SRDCR_VOS_SCALE_2,
+		PWR_SRDCR_VOS_SCALE_3,
+	};
+	static const uint8_t d3cr_vos_values[] = {
+		PWR_D3CR_VOS_SCALE_0,
+		PWR_D3CR_VOS_SCALE_1,
+		PWR_D3CR_VOS_SCALE_2,
+		PWR_D3CR_VOS_SCALE_3,
+	};
+	cm3_assert(scale != PWR_VOS_SCALE_UNDEFINED); 	/* Make sure this has been set. */
 
-	/* Per the manual, VOS0 is implemented as VOS1 + ODEN. Handle this case. */
-	if (scale == PWR_VOS_SCALE_0) {
-		PWR_D3CR = d3cr_masked | PWR_VOS_SCALE_1;
-		SYSCFG_PWRCR |= SYSCFG_PWRCR_ODEN;
+	/* "SmartRun Domain" devices (presently only know of A3/B3/B0) have different mapping.
+	 * Note: DBGMCU_IDCODE_DEV_ID_STM32H7A3 covers all three of these models.
+	 */
+	uint32_t devid = DBGMCU_IDCODE & DBGMCU_IDCODE_DEV_ID_MASK;
+	if (devid == DBGMCU_IDCODE_DEV_ID_STM32H7A3) {
```

Comment:
I see why you're doing this, but this sort of stuff is internal to this file, you don't want to go adding all those id codes to the global file.